pkg/client/oci/oci.go

```diff
@@ -354,9 +425,85 @@ func (r *Repo) Reload() error {
 	return errors.Errorf("reload method is not supported yet")
 }
 
-func (r *Repo) newDockerResolver() remotes.Resolver {
+
+// OciReference contains info about an oci reference
+type ociReference struct {
+	path string
+	tag string
+}
+
+// splitOciReference split the tag or digest from an oci reference if exists
+func splitOciReference(ociRef string) ociReference {
```

Comment:
we probably don't want to add another dependency but in relok8s we use this library to manipulate OCI references, in case it's useful https://pkg.go.dev/github.com/google/go-containerregistry@v0.7.0/pkg/name#ParseReference

Reply:
if there is existing libs for that, I'd use them